src/components/Circles.vue

```diff
@@ -0,0 +1,62 @@
+<template>
+  <div>
+    <ul class="circles" ref="left-circles" style="left: 0"></ul>
+    <ul class="circles" ref="right-circles" style="right: 0"></ul>
+  </div>
+</template>
+
+<script>
+import Shape from "@/components/Shape";
+import Vue from "vue";
+
+export default {
+  name: "CirclesLeft",
+  mounted() {
+    const ComponentClass = Vue.extend(Shape);
+    const leftCircles = this.$refs["left-circles"];
+    const rightCircles = this.$refs["right-circles"];
+
+    function deleteShape(instance, delay, isLeft) {
+      setTimeout(() => {
+        if (isLeft) leftCircles.removeChild(instance.$el);
+        else rightCircles.removeChild(instance.$el);
+        instance.$destroy();
+      }, delay);
+    }
+
+    function addShape() {
```

Comment:
this is a bit against the vue pattern, maybe try to move the functions into
`methods`